stat/distuv/alphastable_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright ©2016 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+	"sort"
+	"testing"
+
+	"golang.org/x/exp/rand"
+	"gonum.org/v1/gonum/stat"
+)
+
+func TestAlphaStable(t *testing.T) {
+	t.Parallel()
+	src := rand.New(rand.NewSource(1))
+	for i, dist := range []AlphaStable{
+		{Alpha: 0.5, Beta: 0, Scale: 1, Mu: 0, Src: src},
+		{Alpha: 1, Beta: 0, Scale: 1, Mu: 0, Src: src},
+		{Alpha: 2, Beta: 0, Scale: 1, Mu: 0, Src: src},
+		{Alpha: 0.5, Beta: 1, Scale: 1, Mu: 0, Src: src},
+		{Alpha: 1, Beta: 1, Scale: 1, Mu: 0, Src: src},
+		{Alpha: 2, Beta: 1, Scale: 1, Mu: 0, Src: src},
+		{Alpha: 0.5, Beta: 0, Scale: 1, Mu: 1, Src: src},
+		{Alpha: 1, Beta: 0, Scale: 1, Mu: 1, Src: src},
+		{Alpha: 2, Beta: 0, Scale: 1, Mu: 1, Src: src},
+		{Alpha: 0.5, Beta: 0.5, Scale: 1, Mu: 1, Src: src},
+		{Alpha: 1, Beta: 0.5, Scale: 1, Mu: 1, Src: src},
+		{Alpha: 1.1, Beta: 0.5, Scale: 1, Mu: 1, Src: src},
+		{Alpha: 2, Beta: 0.5, Scale: 1, Mu: 1, Src: src},
+	} {
+		testAlphaStableAnalytic(t, i, dist)
+	}
+}
+
+func TestAlphaStability(t *testing.T) {
+	t.Parallel()
+	const (
+		n     = 10000
+		ksTol = 2e-2
+	)
+	for i, test := range []struct {
+		alpha, beta1, beta2, scale1, scale2, mu1, mu2 float64
+	}{
+		{2, 0, 0, 1, 2, 0.5, 0.25},
+		{2, 0.9, -0.4, 1, 2, 0.5, 0.25},
+		{1.9, 0, 0, 1, 2, 0.5, 0.25},
+		{1, 0, 0, 1, 2, 0.5, 0.25},
+		{1, -0.5, 0.5, 1, 2, 0.5, 0.25},
+		{0.5, 0, 0, 1, 2, 0.5, 0.25},
+		{0.5, -0.5, 0.5, 1, 2, 0.5, 0.25},
+	} {
+		testStability(t, i, n, test.alpha, test.beta1, test.beta2, test.scale1, test.scale2, test.mu1, test.mu2, ksTol)
+	}
+}
+
+func TestAlphaStableGaussian(t *testing.T) {
+	t.Parallel()
+	src := rand.New(rand.NewSource(1))
+	d := AlphaStable{Alpha: 2, Beta: 0, Scale: 1.5, Mu: -0.4, Src: src}
+	n := 100000
+	x := make([]float64, n)
+	for i := 0; i < n; i++ {
+		x[i] = d.Rand()
+	}
+	checkSkewness(t, 0, x, d, 1e-2)
+	checkExKurtosis(t, 0, x, d, 1e-2)
+	checkMean(t, 0, x, d, 1e-2)
+	checkVarAndStd(t, 0, x, d, 1e-2)
+	checkMode(t, 0, x, d, 5e-2, 1e-1)
+}
+
+func TestAlphaStableMean(t *testing.T) {
+	t.Parallel()
+	src := rand.New(rand.NewSource(1))
+	d := AlphaStable{Alpha: 1.75, Beta: 0.2, Scale: 1.2, Mu: 0.3, Src: src}
+	n := 100000
+	x := make([]float64, n)
+	for i := 0; i < n; i++ {
+		x[i] = d.Rand()
+	}
+	checkMean(t, 0, x, d, 1e-2)
+}
+
+func TestAlphaStableCauchy(t *testing.T) {
```

Comment:
s/Cauchy/Mode/?

Reply:
It only tests Mode for the Cauchy distribution (because there is nothing else to test).